File: LeetCode/88. Merge Sorted Array/solution.py
def merge(nums1: list, m: int, nums2: list, n: int) -> None:
    """
    Do not return anything, modify nums1 in-place instead.
    """
    start = m
    while start < len(nums1):
        if nums1[start] == 0: break
        start += 1
    
    nums1[start:] = nums2[:]
    nums1[:] = sorted(nums1[:])


def merge_shorter(nums1: list, m: int, nums2: list, n: int) -> None:
    nums1[m:] = nums2[:n]
    nums1.sort()

l1 = [1,2,3,0,0,0]
m = 3
l2 = [2,5,6]
n = 3
merge_shorter(l1,m,l2,n)
print(l1)

File: LeetCode/88. Merge Sorted Array/solution.cpp
#include <algorithm>
#include <iostream>
#include <vector>
#include "../prettyprint.h"

using namespace std;

void merge(vector<int> &nums1, int m, vector<int> &nums2, int n) {
	std::copy(nums2.begin(), nums2.end(), nums1.begin() + m);
	std::sort(nums1.begin(), nums1.end());
}

int main() {
	vector<int> l1{1, 2, 3, 0, 0, 0};
	int m = 3;
	vector<int> l2{2, 5, 6};
	int n = 3;

	merge(l1, m, l2, n);

    std::cout << l1 << std::endl;
}